e2e/e2e-tests.js

```diff
@@ -0,0 +1,98 @@
+// @ts-check
+import { repoRoot, run } from "../eng/scripts/helpers.js";
+import { join } from "path";
+import { existsSync, readdirSync, rmSync, writeFileSync } from "fs";
+
+const e2eTestDir = join(repoRoot, "e2e");
+const cadlCmd = process.platform === "win32" ? "cadl.cmd" : "cadl";
+
+function main() {
+  const packages = packPackages();
+  installCompiler(packages["@cadl-lang/compiler"]);
+  testBasicLatest();
+  testBasicCurrentTgz(packages);
+}
+main();
+
+function packPackages() {
+  run("rush", ["publish", "--publish", "--pack", "--include-all"]);
+  const outputFolder = join(repoRoot, "common/temp/artifacts/packages");
+  const files = readdirSync(outputFolder);
+  console.log("Built packages:", files);
+
+  function resolvePacakge(start) {
+    return join(
+      outputFolder,
+      files.find((x) => x.startsWith(start))
+    );
+  }
+
+  return {
+    "@cadl-lang/compiler": resolvePacakge("cadl-lang-compiler-"),
+    "@cadl-lang/openapi3": resolvePacakge("cadl-lang-openapi3-"),
+    "@cadl-lang/rest": resolvePacakge("cadl-lang-rest-"),
+  };
+}
+
+function installCompiler(compilerTgz) {
```

Comment:
This has side effects on the environment (i.e. adding a global command), I wonder if instead we should start from setting up node_modules in the new directory and running the local version of the compiler using npx?  I suppose that this might mask issues with compiler dependencies in the extension modules

Reply:
hhm, didn't think of that but seeems liek just doing `npx <path-to-package.tgz` compile .` works fine, that way we let npx mange the temporary installation